ledger/participant-integration-api/src/main/scala/platform/usermanagement/PersistentUserManagementStore.scala

```diff
@@ -0,0 +1,259 @@
+  UserManagementBasedAuthorizer,
+  UserManagementBasedAuthorizerImpl,
+}
+import com.daml.ledger.api.domain
+import com.daml.ledger.participant.state.index.impl.inmemory.InMemoryUserManagementStore
+import com.daml.ledger.participant.state.index.v2.UserManagementStore
+import com.daml.ledger.participant.state.index.v2.UserManagementStore.{
+  Result,
+  TooManyUserRights,
+  UserExists,
+  UserNotFound,
+  Users,
+}
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Ref.UserId
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Metrics
+import com.daml.platform.apiserver.services.ApiVersionService
+import com.daml.platform.store.appendonlydao.DbDispatcher
+import com.daml.platform.store.backend.UserManagementStorageBackend
+import com.daml.platform.store.backend.common.UserManagementStorageBackendTemplate
+import com.daml.platform.usermanagement.PersistentUserManagementStore.TooManyUserRightsRuntimeException
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object PersistentUserManagementStore {
+
+  /** Intended to be thrown within a DB transaction to abort it.
+    * The resulting failed future will get mapped to a successful future containing scala.util.Left
+    */
+  final case class TooManyUserRightsRuntimeException(userId: Ref.UserId) extends RuntimeException
+
+  def cached(dbDispatcher: DbDispatcher, metrics: Metrics)(implicit
+      executionContext: ExecutionContext
+  ): UserManagementStore = {
+    new CachedUserManagementStore(
+      delegate = new PersistentUserManagementStore(
+        dbDispatcher = dbDispatcher,
+        metrics = metrics,
+        maxNumberOfUserRightsPerUserLimit = ApiVersionService.MaxNumberOfUserRightsPerUser,
+      ),
+      expiryAfterWriteInSeconds = CachedUserManagementStore.ExpiryAfterWriteInSeconds,
+    )
+  }
+
+  def cachedAuthorizer(userManagementStore: UserManagementStore)(implicit
+      executionContext: ExecutionContext,
+      loggingContext: LoggingContext,
+  ): UserManagementBasedAuthorizer = {
+    new CachedUserManagementBasedAuthorizer(
+      delegate = new UserManagementBasedAuthorizerImpl(
+        userManagementStore = userManagementStore
+      )(
+        executionContext = executionContext,
+        loggingContext = loggingContext,
+      ),
+      expiryAfterWriteInSeconds = CachedUserManagementStore.ExpiryAfterWriteInSeconds,
+    )
+  }
+}
+
+class PersistentUserManagementStore(
+    dbDispatcher: DbDispatcher,
+    metrics: Metrics,
+    createAdminUser: Boolean = true,
+    maxNumberOfUserRightsPerUserLimit: Int,
+) extends UserManagementStore {
+
+  private val backend: UserManagementStorageBackend = UserManagementStorageBackendTemplate
+  private val logger = ContextualizedLogger.get(this.getClass)
+
+  implicit private val loggingContext: LoggingContext = LoggingContext.newLoggingContext(identity)
+
+  // TODO participant user management: This is probably not the best place/time to create admin user
```

Comment:
I agree, it should be created in the migration script

Reply:
Ok